src/main/java/seedu/address/model/tag/DormTag.java

```diff
@@ -0,0 +1,55 @@
+package seedu.address.model.tag;
+
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Dorm Tag in Dormie.
+ */
+public class DormTag extends Tag {
+
+    public static final String MESSAGE_CONSTRAINTS = "Dorm Tag should be a non-empty string";
+    public static final String VALIDATION_REGEX = "^(?!\\s*$).+";
```

Comment:
This regex supposes to mean "a non-empty string", I'm not familiar with regex, so feel free to correct me if something is wrong.

Reply:
Could just use ".+" I think.
. means any character
+ means 1 or more times

Reply:
Hmm looks like they are different. 

I think ".+" will only catch "\n", and it will let " " (a string with only space) to pass, which is not intended since we specify that it should be non-empty string.

Reply:
"^[^\\s].+" should work then.
The original is fine though... It's just a little weird to use negative lookahead??

Reply:
Idk hahah, suggested by ChatGPT